core/market/decentralized/src/negotiation/requestor.rs

```diff
@@ -194,15 +214,113 @@ impl RequestorBroker {
         Ok(id)
     }
 
+    pub async fn wait_for_approval(
+        &self,
+        id: &AgreementId,
+        timeout: f32,
+    ) -> Result<ApprovalStatus, WaitForApprovalError> {
+        // TODO: Check if we are owner of Proposal
+        // TODO: What to do with 2 simultaneous calls to wait_for_approval??
+        //  should we reject one? And if so, how to discover, that to calls were made?
+        // TODO: Do we have race conditions here, between checking Agreement state and
+        //  waiting for notifications? Event could come, before we start waiting and
+        //  it will be lost.
```

Comment:
As far as I understand we are using event only to be sure that the `Agreement.state` is already properly set in DB (ie. it is not `Pending`), we are safe here, because the `loop` below will return immediately after querying DB.